### PR TITLE
Add ClassifAI v3 migration guide

### LINKED ISSUE #674: Add a v2-v3 migration guide
### Is your enhancement related to a problem? Please describe.

There's some massive changes coming in v3, some of them breaking changes, and we should put together a migration guide that details all of these changes and what developers may need to change as part of this update.

Thinking things like hook names, REST endpoint changes, changes to various classes if someone happens to be extending those, etc.

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by iamdharmesh**
### Description of the Change
PR adds v3 migration guide docs to help users to upgrade to classifAI v3.

Closes #674 

### Credits
Props @iamdharmesh 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: hookdocs/migration-guide-v2-to-v3.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+**ClassifAI 3.0** introduces significant improvements and restructuring aimed at enhancing flexibility and extensibility. The key changes include a transition to a "**Feature first**" settings approach, where settings screens are now organized around specific Features rather than Service Providers. This allows for the integration of multiple AI providers for each feature, offering users greater customization options.
+
+The migration process is streamlined with an **automated migration routine** that seamlessly transitions your existing settings to the new version upon upgrading to version 3.0.
+
+This migration guide provides a detailed overview of the changes, including updates to the REST API endpoints and information regarding the removed hooks. Explore the details below to smoothly navigate and leverage the enhanced capabilities of ClassifAI 3.0.
+
+### Provider Class Changes
```
Comment (iamdharmesh) [inline hookdocs/migration-guide-v2-to-v3.md:7]: @dkotter, I believe mentioning the updates regarding the restructuring of the provider class is sufficient here. In-depth details, such as specific methods that have been updated or migrated, are not required. What do you think?

Comment (dkotter) [reply hookdocs/migration-guide-v2-to-v3.md:7]: This sounds good to me

**Comment by dkotter** [comment]
@iamdharmesh I've made a handful of changes, most importantly I've added some snippets that show how a third-party developer could add new Features or Providers. In putting this together, I found a few issues with the way we register those that was preventing that from working, so that has been fixed in this PR as well.

**Comment by iamdharmesh** [comment]
Thanks for making changes @dkotter. All looks good to me. The table layout looks like this to me in Chrome browser, may need some CSS updates. is it same for you? Thank you.

image

**Comment by dkotter** [comment]
> Thanks for making changes @dkotter. All looks good to me. The table layout looks like this to me in Chrome browser, may need some CSS updates. is it same for you? Thank you.

Yeah, I added `overflow: scroll` on the table cells because otherwise the table was overlapping the navigation menu on the right. I'm open to other suggestions on how best to handle this but personally I'm fine with the scrolling

**Comment by jeffpaul** [comment]
Agreed, for now, that scrolling cell is acceptable for a dev docs site.

**Pull request merged**